.github/workflows/deploy.yml

```
@@ -0,0 +1,153 @@
+        run: |
+          echo "Rolling back deployment..."
+          kubectl rollout undo deployment/meridian -n staging
+
+  deploy-production:
+    name: Deploy to Production
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/v') || (github.event_name == 'workflow_dispatch' && github.event.inputs.environment == 'production')
+    environment:
+      name: production
+      url: https://meridian.example.com
+    needs: []  # Remove staging dependency if deploying directly to prod
```

actionlint (1.7.8)

"needs" section should not be empty

(syntax-check)